### PR TITLE
ref(device-context): Infer screen solution data

### LINKED ISSUE #26005: Infer `screen_height_pixels` and `screen_width_pixels` from `screen_resolution`
## Summary

The [device context](https://develop.sentry.dev/sdk/event-payloads/contexts/#device-context) of an event has field for the screen size. 
If `screen_resolution` is given `screen_height_pixels` and `screen_width_pixels` should be inferred from it.
If `screen_height_pixels` and `screen_width_pixels` are given `screen_resolution` should be inferred from it.

Also if `screen_resolution` matches a resolution with a known name like `3840 x 2160` for `4K UHD` it should be written next to it.
More examples on [Wikipedia](https://en.wikipedia.org/wiki/Display_resolution#Common_display_resolutions).

## Motivation

I'm always confused it it's width times height or the other way around. So if it's explicitly stated it really helps.

## Additional Context

None

### PR DESCRIPTION
**Pull request by priscilawebdev**
closes: https://github.com/getsentry/sentry/issues/26005

### CODE REVIEW

File: static/app/components/events/contexts/device/types.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ export type DeviceData = {
   online?: boolean;
   screen_density?: number;
   screen_dpi?: number;
-  screen_height_pixels?: string;
+  screen_height_pixels?: number;
```
Comment (priscilawebdev) [inline static/app/components/events/contexts/device/types.tsx:63]: this type was wrong

File: static/app/components/events/contexts/device/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import {defined, formatBytesBase2} from 'app/utils';
+
+import {DeviceData, DeviceKnownDataType} from './types';
+
+export function formatMemory(
```
Comment (priscilawebdev) [inline static/app/components/events/contexts/device/utils.tsx:5]: moved this function from  `formatMemory.tsx ` to this file. Nothing changed

File: static/app/components/events/contexts/device/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    free_memory
+  )}`;
+  if (Number.isInteger(usable_memory) && usable_memory > 0) {
+    memory = `${memory} / Usable: ${formatBytesBase2(usable_memory)}`;
+  }
+
+  return memory;
+}
+
+export function formatStorage(
```
Comment (priscilawebdev) [inline static/app/components/events/contexts/device/utils.tsx:29]: moved this function from `formatStorage.tsx` to this file. Nothing changed

File: static/app/components/events/contexts/device/getDeviceKnownData.tsx [hunk 4]
```diff
@@ -6,7 +6,7 @@ import {defined} from 'app/utils';
 import getDeviceKnownDataDetails from './getDeviceKnownDataDetails';
 import {DeviceData, DeviceKnownDataType} from './types';
 
-function getOperatingSystemKnownData(
+function getDeviceKnownData(
```
Comment (priscilawebdev) [inline static/app/components/events/contexts/device/getDeviceKnownData.tsx:9]: this name was wrong

**Review by markstory — COMMENTED** [review]
Looks good to me. Is this context covered in our existing visual snapshots? If not it would be good to add it.

**Comment by priscilawebdev** [comment]
@markstory I added a new jest snapshot 😉

**Review by dashed — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Pull request merged**